yt/yt/server/node/exec_node/job_environment.cpp

```diff
@@ -75,22 +77,42 @@ class TProcessJobEnvironmentBase
         , Bootstrap_(bootstrap)
     { }
 
-    void Init(int slotCount, double cpuLimit, double idleCpuFraction) override
+    TFuture<void> Init(int slotCount, double cpuLimit, double idleCpuFraction) override
     {
         VERIFY_THREAD_AFFINITY(JobThread);
 
         Bootstrap_->SubscribePopulateAlerts(
             BIND(&TProcessJobEnvironmentBase::PopulateAlerts, MakeWeak(this)));
         // Shutdown all possible processes.
         try {
-            DoInit(slotCount, cpuLimit, idleCpuFraction);
+            return DoInit(slotCount, cpuLimit,idleCpuFraction)
+                .WithTimeout(AsyncInitTimeout);
         } catch (const std::exception& ex) {
             auto error = TError("Failed to clean up processes during initialization")
                 << ex;
             Disable(error);
+            return MakeFuture(error);
         }
     }
 
+    TFuture<void> InitSlot(int slotIndex) override
+    {
+        auto slotInitFuture = BIND([this, slotIndex] {
```

Comment:
this can be simplified with `BIND(&IJobEnvironment::CleanProcesses, MakeStrong(this), slotIndex)`